Sources/Fluent/SQL/GeneralSQLSerializer.swift

```diff
@@ -403,8 +403,17 @@ open class GeneralSQLSerializer: SQLSerializer {
 
     open func sql(_ type: Schema.Field.DataType) -> String {
         switch type {
-        case .id:
-            return "INTEGER PRIMARY KEY"
+        case .id(let keyType):
+            switch keyType ?? Schema.Field.KeyType.int {
+            case .int:
+                return "INTEGER PRIMARY KEY"
+                
+            case .string(let length):
+                return "VARCHAR(\(length ?? 128)) PRIMARY KEY"
```

Comment:
Pointing this out too as it may be controversial: I changed from STRING to VARCHAR because this would work across MySQL, PostgreSQL, SQLite at least whereas STRING would not I'm pretty sure.